overlord/snapstate/handlers.go

```diff
@@ -871,7 +871,10 @@ func (m *SnapManager) doPreDownloadSnap(t *state.Task, tomb *tomb.Tomb) error {
 		return err
 	}
 
-	if err := softCheckNothingRunningForRefresh(st, snapst, snapsup, info); err != nil {
+	lock, err := m.backend.RunInhibitSnapForUnlink(info, runinhibit.HintInhibitedForPreDownload, func() error {
+		return inhibitRefresh(st, snapst, snapsup, info)
```

Comment:
this does things with snapst and also potentially notify that we have run out which I'm a bit unsure we want from here? should we call refreshAppsCheck directly instead?